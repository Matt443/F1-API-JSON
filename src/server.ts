import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}