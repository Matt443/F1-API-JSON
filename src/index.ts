import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
