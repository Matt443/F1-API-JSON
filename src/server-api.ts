import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
