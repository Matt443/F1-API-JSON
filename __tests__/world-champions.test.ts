import { getWorldChampions } from "../src/server";

test("hall of fame", async () => {
    expect(await getWorldChampions()).toMatchSnapshot();
});
