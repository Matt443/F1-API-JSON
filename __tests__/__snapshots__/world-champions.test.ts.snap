// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hall of fame 1`] = `
[
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-856948472.webp",
    "name": "Nino Farina",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9North/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-827612408.webp",
    "name": "Juan Manuel Fangio",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-106501211.webp",
    "name": "Alberto Ascari",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9North/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-1068321752.webp",
    "name": "Mike Hawthorn",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9North/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-3323162.webp",
    "name": "Jack Brabham",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9North/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-71183243.webp",
    "name": "Phil Hill",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-864209082.webp",
    "name": "Graham Hill",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-1328924500.webp",
    "name": "Jim Clark",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-3072443.webp",
    "name": "John Surtees",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-1063718326.webp",
    "name": "Denny Hulme",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-1443308803.webp",
    "name": "Jackie Stewart",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-831631668.webp",
    "name": "Jochen Rindt",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-1189103543.webp",
    "name": "Emerson Fittipaldi",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-525577811.webp",
    "name": "Niki Lauda",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-827647160.webp",
    "name": "James Hunt",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-851859388.webp",
    "name": "Mario Andretti",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-79750356.webp",
    "name": "Jody Scheckter",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-868507192.webp",
    "name": "Alan Jones",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-1066217104.webp",
    "name": "Nelson Piquet",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-464497739.webp",
    "name": "Keke Rosberg",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-537112453.webp",
    "name": "Alain Prost",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-659224375.webp",
    "name": "Ayrton Senna",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-865701928.webp",
    "name": "Nigel Mansell",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-72220757.webp",
    "name": "Michael Schumacher",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-868558648.webp",
    "name": "Damon Hill",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-52918273.webp",
    "name": "Jacques Villeneuve",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-647968497.webp",
    "name": "Mika Hakkinen",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-55768339.webp",
    "name": "Fernando Alonso",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9South/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-2071874459.webp",
    "name": "Kimi Raikkonen",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-1285771774.webp",
    "name": "Lewis Hamilton",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-91979303.webp",
    "name": "Jenson Button",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9Centre/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-82801122.webp",
    "name": "Sebastian Vettel",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/t_16by9North/c_lfill,w_1006/q_auto/v1740000000/fom-website/manual/Hall%20of%20Fame%202024/GettyImages-607523452.webp",
    "name": "Nico Rosberg",
  },
  {
    "driver_image": "https://media.formula1.com/image/upload/c_lfill,w_1006/q_auto/v1740000000/content/dam/fom-website/sutton/2021/AbuDhabi/Sunday/1358623736.webp",
    "name": "Max Verstappen",
  },
]
`;
